iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/SecureValuesExposureCheckTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.checks;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.common.testing.Verifier;
+
+class SecureValuesExposureCheckTest {
+  @Test
+  void testJson() {
+    ArmVerifier.verify("SecureValuesExposureCheck/Microsoft.Resources_deployments_compliant_secure_scope.json", new SecureValuesExposureCheck());
+    ArmVerifier.verify("SecureValuesExposureCheck/Microsoft.Resources_deployments_compliant_no_top_level_parameter.json", new SecureValuesExposureCheck());
+    ArmVerifier.verify("SecureValuesExposureCheck/Microsoft.Resources_deployments_noncompliant.json", new SecureValuesExposureCheck(),
+      Verifier.issue(12, 14, 12, 47, "Change this code to not use an outer expression evaluation scope in nested templates."));
+  }
+
+  @Test
+  void testBicep() {
+    BicepVerifier.verifyNoIssue("SecureValuesExposureCheck/Microsoft.Resources_deployments_compliant.bicep", new SecureValuesExposureCheck());
+    BicepVerifier.verifyNoIssue("SecureValuesExposureCheck/Microsoft.Resources_deployments_compliant_no_usages.bicep", new SecureValuesExposureCheck());
+    BicepVerifier.verify("SecureValuesExposureCheck/Microsoft.Resources_deployments_noncompliant.bicep", new SecureValuesExposureCheck());
+    // TODO: files `main.bicep` and `vm.bicep` are a preferred fix for the issue, but we don't support multiple files analysis yet
```

Comment:
You could write a different test method for main.bicep and vm.bicep, and disable this test with @disabled annotation with text parameters your todo.